### ISSUE TITLE
List from_be_bytes as Alternative in Transmute Docs

### ISSUE THREAD

**Issue by domenukk** [rust-lang/rust#71187]
Many code bases I see use `transmute` to convert raw bytes (`&[u8]`) to `u32`, `f64`, etc. - while they could, and probably should, use `from_le_bytes` or `from_be_bytes`.
The `transmute` docs include a lengthy section about Alternatives, but this obvious one seems to be missing.

**Comment by huangjiahua** [comment]
@domenukk I've used `from_le_bytes` and `from_be_bytes` several times in my little project. I want to have a try. Will this code snippet work?
```rust
    let raw_data = &[0x78, 0x56, 0x34, 0x12];
    
    let num = unsafe {
        std::mem::transmute::<[u8; 4], u32>(*raw_data)
    };
    // This result depends on your machine
    assert_eq!(num, 0x12345678);


    // use from_be_bytes and from_le_bytes instead to produce predictable result
    let num = u32::from_le_bytes(*raw_data);
    assert_eq!(num, 0x12345678);
    
    let num = u32::from_be_bytes(*raw_data);
    assert_eq!(num, 0x78563412);
```

**Comment by Duddino** [comment]
@huangjiahua maybe you can also add `u32::from_ne_bytes()` as an alternative to `transmute` but without unsafe

**Comment by Alexendoo** [comment]
Closed by #71315
